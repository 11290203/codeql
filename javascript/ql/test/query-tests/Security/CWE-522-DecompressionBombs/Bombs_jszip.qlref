experimental/Security/CWE-522-DecompressionBombs/Bombs_jszip.ql