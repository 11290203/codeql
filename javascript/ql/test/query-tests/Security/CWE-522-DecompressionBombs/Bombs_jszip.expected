nodes
| YargsTests1.js:3:5:3:62 | argv1 |
| YargsTests1.js:3:13:3:62 | require ... )).argv |
| YargsTests1.js:3:13:3:62 | require ... )).argv |
| YargsTests1.js:8:14:8:18 | argv1 |
| YargsTests1.js:8:14:8:20 | argv1.a |
| YargsTests1.js:8:14:8:20 | argv1.a |
| argvTests1.js:4:14:4:20 | argv[2] |
| argvTests1.js:4:14:4:20 | argv[2] |
| argvTests1.js:6:15:6:17 | val |
| argvTests1.js:6:15:6:17 | val |
| argvTests1.js:7:18:7:20 | val |
| commanderTests1.js:5:18:5:22 | value |
| commanderTests1.js:5:18:5:22 | value |
| commanderTests1.js:5:25:5:32 | previous |
| commanderTests1.js:5:25:5:32 | previous |
| commanderTests1.js:6:18:6:25 | previous |
| commanderTests1.js:7:25:7:29 | value |
| commanderTests1.js:15:14:15:28 | options.collect |
| commanderTests1.js:15:14:15:28 | options.collect |
| commanderTests2.js:10:22:10:40 | this.opts().zipPath |
| commanderTests2.js:10:22:10:40 | this.opts().zipPath |
| commanderTests2.js:11:22:11:30 | this.args |
| commanderTests2.js:11:22:11:30 | this.args |
| commanderTests2.js:11:22:11:33 | this.args[0] |
| commanderTests3.js:10:18:10:32 | options.zipPath |
| commanderTests3.js:10:18:10:32 | options.zipPath |
| commanderTests3.js:16:14:16:20 | zipPath |
| commanderTests3.js:16:14:16:20 | zipPath |
| commanderTests3.js:16:23:16:30 | password |
| commanderTests3.js:16:23:16:30 | password |
| commanderTests3.js:17:22:17:28 | zipPath |
| commanderTests3.js:18:22:18:29 | password |
| commanderTests3.js:25:23:25:26 | dirs |
| commanderTests3.js:25:23:25:26 | dirs |
| commanderTests3.js:26:9:26:12 | dirs |
| commanderTests3.js:26:23:26:29 | zipPath |
| commanderTests3.js:26:23:26:29 | zipPath |
| commanderTests3.js:27:26:27:32 | zipPath |
| commanderTests3.js:34:26:34:51 | thisCom ... zipPath |
| commanderTests3.js:34:26:34:51 | thisCom ... zipPath |
| main-jszip.js:13:22:13:24 | req |
| main-jszip.js:13:22:13:24 | req |
| main-jszip.js:14:24:14:26 | req |
| main-jszip.js:14:24:14:32 | req.files |
| main-jszip.js:14:24:14:32 | req.files |
| main-jszip.js:14:24:14:44 | req.fil ... ombFile |
| main-jszip.js:15:13:15:23 | tmpObj["a"] |
| main-jszip.js:35:18:35:24 | zipFile |
| main-jszip.js:35:18:35:24 | zipFile |
| main-jszip.js:35:18:35:24 | zipFile |
| main-jszip.js:36:22:36:28 | zipFile |
| main-jszip.js:36:22:36:28 | zipFile |
| main-jszip.js:36:22:36:33 | zipFile.data |
| main-jszip.js:36:22:36:33 | zipFile.data |
| main-jszip.js:36:22:36:33 | zipFile.data |
| main-jszip.js:47:23:47:26 | path |
| main-jszip.js:47:23:47:26 | path |
| main-jszip.js:47:23:47:26 | path |
| main-jszip.js:48:17:48:20 | path |
| main-jszip.js:48:17:48:20 | path |
| main-jszip.js:49:26:49:29 | data |
| main-jszip.js:49:26:49:29 | data |
| main-jszip.js:49:26:49:29 | data |
| main-jszip.js:51:29:51:32 | data |
| main-jszip.js:51:29:51:32 | data |
| main-jszip.js:51:29:51:32 | data |
edges
| YargsTests1.js:3:5:3:62 | argv1 | YargsTests1.js:8:14:8:18 | argv1 |
| YargsTests1.js:3:13:3:62 | require ... )).argv | YargsTests1.js:3:5:3:62 | argv1 |
| YargsTests1.js:3:13:3:62 | require ... )).argv | YargsTests1.js:3:5:3:62 | argv1 |
| YargsTests1.js:8:14:8:18 | argv1 | YargsTests1.js:8:14:8:20 | argv1.a |
| YargsTests1.js:8:14:8:20 | argv1.a | main-jszip.js:47:23:47:26 | path |
| YargsTests1.js:8:14:8:20 | argv1.a | main-jszip.js:47:23:47:26 | path |
| argvTests1.js:4:14:4:20 | argv[2] | main-jszip.js:47:23:47:26 | path |
| argvTests1.js:4:14:4:20 | argv[2] | main-jszip.js:47:23:47:26 | path |
| argvTests1.js:6:15:6:17 | val | argvTests1.js:7:18:7:20 | val |
| argvTests1.js:6:15:6:17 | val | argvTests1.js:7:18:7:20 | val |
| argvTests1.js:7:18:7:20 | val | main-jszip.js:47:23:47:26 | path |
| commanderTests1.js:5:18:5:22 | value | commanderTests1.js:7:25:7:29 | value |
| commanderTests1.js:5:18:5:22 | value | commanderTests1.js:7:25:7:29 | value |
| commanderTests1.js:5:25:5:32 | previous | commanderTests1.js:6:18:6:25 | previous |
| commanderTests1.js:5:25:5:32 | previous | commanderTests1.js:6:18:6:25 | previous |
| commanderTests1.js:6:18:6:25 | previous | main-jszip.js:47:23:47:26 | path |
| commanderTests1.js:7:25:7:29 | value | main-jszip.js:47:23:47:26 | path |
| commanderTests1.js:15:14:15:28 | options.collect | main-jszip.js:47:23:47:26 | path |
| commanderTests1.js:15:14:15:28 | options.collect | main-jszip.js:47:23:47:26 | path |
| commanderTests2.js:10:22:10:40 | this.opts().zipPath | main-jszip.js:47:23:47:26 | path |
| commanderTests2.js:10:22:10:40 | this.opts().zipPath | main-jszip.js:47:23:47:26 | path |
| commanderTests2.js:11:22:11:30 | this.args | commanderTests2.js:11:22:11:33 | this.args[0] |
| commanderTests2.js:11:22:11:30 | this.args | commanderTests2.js:11:22:11:33 | this.args[0] |
| commanderTests2.js:11:22:11:33 | this.args[0] | main-jszip.js:47:23:47:26 | path |
| commanderTests3.js:10:18:10:32 | options.zipPath | main-jszip.js:47:23:47:26 | path |
| commanderTests3.js:10:18:10:32 | options.zipPath | main-jszip.js:47:23:47:26 | path |
| commanderTests3.js:16:14:16:20 | zipPath | commanderTests3.js:17:22:17:28 | zipPath |
| commanderTests3.js:16:14:16:20 | zipPath | commanderTests3.js:17:22:17:28 | zipPath |
| commanderTests3.js:16:23:16:30 | password | commanderTests3.js:18:22:18:29 | password |
| commanderTests3.js:16:23:16:30 | password | commanderTests3.js:18:22:18:29 | password |
| commanderTests3.js:17:22:17:28 | zipPath | main-jszip.js:47:23:47:26 | path |
| commanderTests3.js:18:22:18:29 | password | main-jszip.js:47:23:47:26 | path |
| commanderTests3.js:25:23:25:26 | dirs | commanderTests3.js:26:9:26:12 | dirs |
| commanderTests3.js:25:23:25:26 | dirs | commanderTests3.js:26:9:26:12 | dirs |
| commanderTests3.js:26:9:26:12 | dirs | commanderTests3.js:26:23:26:29 | zipPath |
| commanderTests3.js:26:23:26:29 | zipPath | commanderTests3.js:27:26:27:32 | zipPath |
| commanderTests3.js:26:23:26:29 | zipPath | commanderTests3.js:27:26:27:32 | zipPath |
| commanderTests3.js:27:26:27:32 | zipPath | main-jszip.js:47:23:47:26 | path |
| commanderTests3.js:34:26:34:51 | thisCom ... zipPath | main-jszip.js:47:23:47:26 | path |
| commanderTests3.js:34:26:34:51 | thisCom ... zipPath | main-jszip.js:47:23:47:26 | path |
| main-jszip.js:13:22:13:24 | req | main-jszip.js:14:24:14:26 | req |
| main-jszip.js:13:22:13:24 | req | main-jszip.js:14:24:14:26 | req |
| main-jszip.js:14:24:14:26 | req | main-jszip.js:14:24:14:32 | req.files |
| main-jszip.js:14:24:14:32 | req.files | main-jszip.js:14:24:14:44 | req.fil ... ombFile |
| main-jszip.js:14:24:14:32 | req.files | main-jszip.js:14:24:14:44 | req.fil ... ombFile |
| main-jszip.js:14:24:14:44 | req.fil ... ombFile | main-jszip.js:15:13:15:23 | tmpObj["a"] |
| main-jszip.js:15:13:15:23 | tmpObj["a"] | main-jszip.js:35:18:35:24 | zipFile |
| main-jszip.js:35:18:35:24 | zipFile | main-jszip.js:36:22:36:28 | zipFile |
| main-jszip.js:35:18:35:24 | zipFile | main-jszip.js:36:22:36:28 | zipFile |
| main-jszip.js:35:18:35:24 | zipFile | main-jszip.js:36:22:36:28 | zipFile |
| main-jszip.js:36:22:36:28 | zipFile | main-jszip.js:36:22:36:33 | zipFile.data |
| main-jszip.js:36:22:36:28 | zipFile | main-jszip.js:36:22:36:33 | zipFile.data |
| main-jszip.js:36:22:36:28 | zipFile | main-jszip.js:36:22:36:33 | zipFile.data |
| main-jszip.js:36:22:36:28 | zipFile | main-jszip.js:36:22:36:33 | zipFile.data |
| main-jszip.js:47:23:47:26 | path | main-jszip.js:48:17:48:20 | path |
| main-jszip.js:47:23:47:26 | path | main-jszip.js:48:17:48:20 | path |
| main-jszip.js:47:23:47:26 | path | main-jszip.js:48:17:48:20 | path |
| main-jszip.js:48:17:48:20 | path | main-jszip.js:49:26:49:29 | data |
| main-jszip.js:48:17:48:20 | path | main-jszip.js:49:26:49:29 | data |
| main-jszip.js:49:26:49:29 | data | main-jszip.js:51:29:51:32 | data |
| main-jszip.js:49:26:49:29 | data | main-jszip.js:51:29:51:32 | data |
| main-jszip.js:49:26:49:29 | data | main-jszip.js:51:29:51:32 | data |
| main-jszip.js:49:26:49:29 | data | main-jszip.js:51:29:51:32 | data |
| main-jszip.js:49:26:49:29 | data | main-jszip.js:51:29:51:32 | data |
| main-jszip.js:49:26:49:29 | data | main-jszip.js:51:29:51:32 | data |
#select
| main-jszip.js:36:22:36:33 | zipFile.data | main-jszip.js:13:22:13:24 | req | main-jszip.js:36:22:36:33 | zipFile.data | This file extraction depends on a $@. | main-jszip.js:13:22:13:24 | req | potentially untrusted source |
| main-jszip.js:36:22:36:33 | zipFile.data | main-jszip.js:14:24:14:32 | req.files | main-jszip.js:36:22:36:33 | zipFile.data | This file extraction depends on a $@. | main-jszip.js:14:24:14:32 | req.files | potentially untrusted source |
| main-jszip.js:36:22:36:33 | zipFile.data | main-jszip.js:35:18:35:24 | zipFile | main-jszip.js:36:22:36:33 | zipFile.data | This file extraction depends on a $@. | main-jszip.js:35:18:35:24 | zipFile | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | YargsTests1.js:3:13:3:62 | require ... )).argv | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | YargsTests1.js:3:13:3:62 | require ... )).argv | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | YargsTests1.js:8:14:8:20 | argv1.a | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | YargsTests1.js:8:14:8:20 | argv1.a | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | argvTests1.js:4:14:4:20 | argv[2] | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | argvTests1.js:4:14:4:20 | argv[2] | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | argvTests1.js:6:15:6:17 | val | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | argvTests1.js:6:15:6:17 | val | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | commanderTests1.js:5:18:5:22 | value | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | commanderTests1.js:5:18:5:22 | value | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | commanderTests1.js:5:25:5:32 | previous | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | commanderTests1.js:5:25:5:32 | previous | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | commanderTests1.js:15:14:15:28 | options.collect | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | commanderTests1.js:15:14:15:28 | options.collect | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | commanderTests2.js:10:22:10:40 | this.opts().zipPath | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | commanderTests2.js:10:22:10:40 | this.opts().zipPath | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | commanderTests2.js:11:22:11:30 | this.args | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | commanderTests2.js:11:22:11:30 | this.args | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | commanderTests3.js:10:18:10:32 | options.zipPath | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | commanderTests3.js:10:18:10:32 | options.zipPath | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | commanderTests3.js:16:14:16:20 | zipPath | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | commanderTests3.js:16:14:16:20 | zipPath | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | commanderTests3.js:16:23:16:30 | password | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | commanderTests3.js:16:23:16:30 | password | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | commanderTests3.js:25:23:25:26 | dirs | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | commanderTests3.js:25:23:25:26 | dirs | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | commanderTests3.js:26:23:26:29 | zipPath | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | commanderTests3.js:26:23:26:29 | zipPath | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | commanderTests3.js:34:26:34:51 | thisCom ... zipPath | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | commanderTests3.js:34:26:34:51 | thisCom ... zipPath | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | main-jszip.js:47:23:47:26 | path | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | main-jszip.js:47:23:47:26 | path | potentially untrusted source |
| main-jszip.js:51:29:51:32 | data | main-jszip.js:49:26:49:29 | data | main-jszip.js:51:29:51:32 | data | This file extraction depends on a $@. | main-jszip.js:49:26:49:29 | data | potentially untrusted source |
